File: JavaSource/src/main/java/ru/akhcheck/patterns/chainresponsibility/handlers/FileHandler.java
package ru.akhcheck.patterns.chainresponsibility.handlers;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardOpenOption;
import java.util.Set;

/** Пишет запись в файл. */
public class FileHandler extends BaseHandler {
    private final Path path;

    public FileHandler(Path path, Set<LogLevel> logLevels) {
        super(logLevels);
        this.path = path;
    }

    @Override
    protected void process(Log log) {
        try {
            Files.writeString(
                    path,
                    log + System.lineSeparator(),
                    StandardCharsets.UTF_8,
                    StandardOpenOption.CREATE,
                    StandardOpenOption.APPEND
            );
        } catch (IOException error) {
            throw new IllegalStateException("cannot write to " + path, error);
        }
    }
}

File: ChainResponsibility/python-source/handlers/FileHandler.py
from __future__ import annotations

from pathlib import Path

from .BaseHandler import BaseHandler
from .Log import Log
from .LogLevel import LogLevel


class FileHandler(BaseHandler):
    """Пишет запись в файл."""

    def __init__(self, filename: str | Path, log_levels: set[LogLevel] | None = None) -> None:
        super().__init__(log_levels)
        self._path = Path(filename)

    def process(self, log: Log) -> None:
        with self._path.open("a", encoding="utf-8") as stream:
            stream.write(str(log) + "\n")
